sdcm/collectd.py

```diff
@@ -0,0 +1,421 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2016 ScyllaDB
+
+import os
+import tempfile
+import shutil
+
+
+class CollectdSetup(object):
+
+    def _implementation_start_collectd_service(self):
+        raise NotImplementedError
+
+    def _implementation_collectd_exporter_setup(self):
+        raise NotImplementedError
+
+    def _implementation_collectd_config(self):
```

Comment:
I don't like the idea of having the implementation functions as private functions, since that leads the specialized classes only implementing private functions. So I'd call `self._implementation_start_collectd_service` as simply `start_collectd_service`, so on and so forth.